sdk/core/core-rest-pipeline/src/util/file.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { isReadableStream } from "./typeGuards";
+import { toWebStream } from "./stream";
+
+/**
+ * Options passed into createFile specifying metadata about the file.
+ */
+export interface CreateFileOptions {
+  /**
+   * The MIME type of the file.
+   */
+  type?: string;
+
+  /**
+   * Last modified time of the file as a UNIX timestamp.
+   * This will default to the current date.
+   */
+  lastModified?: number;
+
+  /**
+   * relative path of this file when uploading a directory.
+   */
+  webkitRelativePath?: string;
+}
+
+/**
+ * Extra options for createFile when a stream is being passed in.
+ */
+export interface CreateFileFromStreamOptions extends CreateFileOptions {
```

Comment:
```suggestion
export interface CreateFileOptions extends CreateFileOptions {
```

Reply:
Just realized that this name already exists. We could keep this name then but it may feel awkward.

Reply:
What's the issue with the name here? `CreateFileOptions` is to `createFile` as `CreateFileFromStreamOptions` is to `createFileFromStream`.

Reply:
isn't a `createFile` overload now?

Reply:
Oh, I didn’t push that change, my bad 🤦‍♂️